### DISCUSSION TITLE
Number of time interval for predcition

### DISCUSSION THREAD

**Discussion by Gritank** [hazdzz/stgcn#22] [Q&A] [answered]
Hi, 
First of all, thanks for the code and the model. That's really a nice job and full of option ! 
I'm currently trying to use it to another purpose, and I try to understand the code and the model correctly. 

I'm a bit confused between the code and the paper explaination. Maybe something that I've not well understood. So my question is that into the paper, the model should predict the H next time steps (in the code the n_pred). So the model should predict by default 3 values for the 3 next time steps. But what I see is that the prediction is made for 1 timestamp but at n_pred ahead. Is that right  ?

**Comment by hazdzz** [comment, marked as answer]
You can rewrite the code to meet your requirements. I haven't worked on this topic for several years, so currently I am only fixing bugs in the code and will not be adding any new features.

**Comment by passion7755** [comment]
> So the model should predict by default 3 values for the 3 next time steps. But what I see is that the prediction is made for 1 timestamp but at n_pred ahead

Hello!

I'm having the same problem as you in reproducing this code. I set n_pred to 6, but the prediction comes out to only 1 time step.
Have you solved this problem perfectly? Can you communicate with me about this issue?
